src/Loop/Driver.php

```diff
@@ -285,4 +311,35 @@
      * @return null|object|resource The loop handle the event loop operates on. `null` if there is none.
      */
     abstract public function getHandle();
+
+    /**
+     * Create a watcher identifier.
+     *
+     * Any watcher identifiers returned to clients by the driver MUST be created using this method. The returned watcher
+     * identifier SHALL be unique for the duration of the PHP process.
+     *
+     * @return string The watcher identifier.
+     */
+    final protected function createWatcherId()
+    {
+        return $this->watcherIdPrefix . $this->nextWatcherId++;
+    }
+
+    /**
+     * Validate a watcher identifier which was passed to the driver by a client.
```

Comment:
I guess that as the invocation of the driver by the client with the invalid watcher has not yet returned, _has been_ would be better. I'll change it. Thanks.